CMakeLists.txt

```diff
@@ -1,177 +1,103 @@
-gtest/src/condition_test_node.cpp
+if(catkin_FOUND)
+  find_package(catkin REQUIRED COMPONENTS )
+
+  message(STATUS "------------------------------------------")
+  message(STATUS "BehaviourTree is being built using CATKIN.")
+  message(STATUS "------------------------------------------")
+
+  catkin_package(
+    INCLUDE_DIRS include
+    LIBRARIES behavior_tree_core
+    )
+else(catkin_FOUND)
+
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/sample)
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+	
+    find_package(GTest)
+    include_directories(${GTEST_INCLUDE_DIRS})
+    if(NOT GTEST_FOUND)
+        message(WARNING " GTest not found!")
+    endif(NOT GTEST_FOUND)
+
+endif(catkin_FOUND)
+
+
+set(BT_Source
+	src/action_node.cpp
+	src/condition_node.cpp
+	src/control_node.cpp
+	src/exceptions.cpp
+	src/leaf_node.cpp
+	src/tick_engine.cpp
+	src/parallel_node.cpp
+	src/fallback_node.cpp
+	src/sequence_node.cpp
+	src/fallback_node_with_memory.cpp
+	src/sequence_node_with_memory.cpp
+	src/tree_node.cpp
 )
+include_directories(include)
 
 
 ######################################################
-# SETTING BINARIES AND LIBRARY OUTPUT LOCATIONS
+# COMPILING LIBRARY
 ######################################################
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/sample)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+add_library(behavior_tree_core STATIC ${BT_Source} )
 
-######################################################
-# COMPILING GTEST
-######################################################
-if(GTEST_FOUND)
-    #add_executable(btpp_gtest gtest/gtest_tree.cpp ${BTSrcLibrary} ${BTHeadLibrary})
-    #target_link_libraries(btpp_gtest  ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES} ${X11_LIBRARIES})
-endif(GTEST_FOUND)
 
 ######################################################
-# COMPILING SAMPLE EXAMPLE
+# TESTS
 ######################################################
-add_executable(btpp_example src/example.cpp ${BTSrcLibrary} ${BTHeadLibrary})
-target_link_libraries(btpp_example  ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${X11_LIBRARIES})
+set(BT_Tests
+        gtest/src/action_test_node.cpp
+        gtest/src/condition_test_node.cpp
+        gtest/gtest_tree.cpp
+)
 
-######################################################
-# COMPILING LIBRARY
-######################################################
-add_library(BTppLib STATIC ${BTSrcLibrary} ${BTHeadLibrary})
-target_link_libraries(BTppLib ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${X11_LIBRARIES})
+if(catkin_FOUND AND CATKIN_ENABLE_TESTING)
+    include_directories(gtest/include)
+
+    catkin_add_gtest(behavior_tree_core_test ${BT_Tests} )
+    target_link_libraries(behavior_tree_core_test
+        behavior_tree_core
+        ${catkin_LIBRARIES} )
+	    
+elseif(GTEST_FOUND)
+    include_directories(gtest/include)
+
+    add_executable(behavior_tree_core_test ${BT_Tests} )
+    target_link_libraries(btpp_test 
```

Comment:
Should this be `target_link_libraries(behavior_tree_core_test ` ?
Probably travis skipped this part because `catkin_FOUND AND CATKIN_ENABLE_TESTING` was true.